sdk/python/tests/compiler/testdata/resourceop_basic.yaml

```diff
@@ -13,90 +13,77 @@
-    description: Kubectl wrapper image
-    name: image
-    type: string
-  - default: 'false'
-    description: Enable set owner reference for created resource.
-    name: set-ownerreference
-    type: string
-  results:
-  - description: '{}'
-    name: manifest
-  - description: '{.metadata.name}'
-    name: name
-  steps:
-  - args:
-    - --action=$(params.action)
-    - --merge-strategy=$(params.merge-strategy)
-    - "--manifest=apiVersion: batch/v1\nkind: Job\nmetadata:\n  generateName: resourceop-basic-job-\n\
-      spec:\n  backoffLimit: 4\n  template:\n    metadata:\n      name: resource-basic\n\
-      \    spec:\n      containers:\n      - command:\n        - /usr/bin/env\n  \
-      \      image: k8s.gcr.io/busybox\n        name: sample-container\n      restartPolicy:\
-      \ Never\n"
-    - --output=$(params.output)
-    - --success-condition=$(params.success-condition)
-    - --failure-condition=$(params.failure-condition)
-    - --set-ownerreference=$(params.set-ownerreference)
-    image: $(params.image)
-    name: main
-    resources: {}
----
-apiVersion: tekton.dev/v1beta1
-kind: Pipeline
+kind: PipelineRun
 metadata:
   annotations:
     pipelines.kubeflow.org/pipeline_spec: '{"description": "A Basic Example on ResourceOp
       Usage.", "name": "ResourceOp Basic"}'
     sidecar.istio.io/inject: 'false'
-  name: resourceop-basic
-spec:
-  tasks:
-  - name: test-step
-    params:
-    - name: action
-      value: create
-    - name: output
-      value: "- name: manifest\n  valueFrom: '{}'\n- name: name\n  valueFrom: '{.metadata.name}'\n"
-    - name: set-ownerreference
-      value: 'false'
-    taskRef:
-      name: test-step
----
-apiVersion: tekton.dev/v1beta1
-kind: PipelineRun
-metadata:
-  annotations:
     tekton.dev/input_artifacts: '{}'
     tekton.dev/output_artifacts: '{}'
-  name: resourceop-basic-run
+  labels:
+    pipelines.kubeflow.org/pipeline-sdk-type: kfp
+  name: resourceop-basic
 spec:
-  pipelineRef:
-    name: resourceop-basic
+  pipelineSpec:
+    tasks:
+    - name: test-step
+      params:
+      - name: action
+        value: create
+      - name: output
+        value: "- name: manifest\n  valueFrom: '{}'\n- name: name\n  valueFrom: '{.metadata.name}'\n"
+      - name: success-condition
+        value: status.succeeded > 0
+      - name: failure-condition
+        value: status.failed > 0
+      - name: set-ownerreference
+        value: 'false'
+      taskSpec:
+        params:
+        - description: Action on the resource
+          name: action
+          type: string
+        - default: strategic
+          description: Merge strategy when using action patch
+          name: merge-strategy
+          type: string
+        - description: An express to retrieval data from resource.
+          name: output
+          type: string
+        - description: A label selector express to decide if the action on resource
+            is success.
+          name: success-condition
```

Comment:
Looks like the task default parameter with empty string all get dropped somewhere, this is probably why the resourceop task was failing with the original test.

Reply:
oh no, the empty string non-default default parameters again :-(

Reply:
I pushed another commit, reinstating the empty string default non value default params

Reply:
It was there in the previous task template, probably one of the clean up functions popped out the empty dictionary value.